python/ray/train/lightning/lightning_trainer.py

```diff
@@ -527,6 +535,13 @@ def _lightning_train_loop_per_worker(config):
     trainer_config["enable_checkpointing"] = True
     model_checkpoint_config["save_last"] = True
 
+    # Convert dirpath to an absolute path under the trial directory
+    ckpt_dirpath = model_checkpoint_config.get("dirpath", None)
+    if ckpt_dirpath and not os.path.isabs(ckpt_dirpath):
```

Comment:
We should also expanduser, if the user passes in something like `"~/checkpoint_dirpath"`:

```python
os.path.isabs(os.path.expanduser(dirpath))
```

Reply:
Make sense : )